src/Hyperion/Report.hs

```diff
@@ -29,7 +30,7 @@ data Report = Report
 makeLenses ''Report
 deriveJSON defaultOptions{ fieldLabelModifier = camelTo2 '_' . drop (length @[] "_report") } ''Report
 
-json :: UTCTime -> Maybe Text -> HashMap Text Report -> JSON.Value
+json :: UTCTime -> Maybe Text -> HashMap (Text, [Int]) Report -> JSON.Value
```

Comment:
This calls for a record, if only to document what is going on here. Tuples are  seldom useful in Haskell.

Taking a step back, what you're doing here is saying that benchmark names aren't just stringly text anymore. They're structured things, that include a base name (of type `Text`) and zero or more benchmark parameters. If that is so, then define a datatype that says just that!

Then, you can amend the `Report` record to say:

```haskell
data Report Report
  { _reportBenchmarkId :: !BenchmarkId
  , ...
  }
```

Better yet, this makes the spurious `uncurry` occurrence in the call to `analyze` disappear, if you make `analyze` take a `BenchmarkId` as an input. That's a win for code readability, because `uncurry` is just adding distracting noise to an otherwise straightforward line in [Main.hs](https://github.com/tweag/hyperion/pull/27/files#diff-c613859236dbc1f34b63c869c01e7f26R172).